server/odc-service/src/main/java/com/oceanbase/odc/service/db/session/DefaultDBSessionManage.java

```diff
@@ -198,33 +210,182 @@ private List<KillSessionResult> doKillSessionOrQuery(
+
+    /**
+     * process the execution result after the first kill commands if the result contains unknown thread
+     * id exception, try to use other solutions to execute the kill commands
+     * 
+     * @param connectionSession
+     * @param results
+     * @return
+     */
+    private List<JdbcGeneralResult> processResults(ConnectionSession connectionSession,
+            List<JdbcGeneralResult> results) {
+        boolean isDirectedOBServer = checkObServerDirected(connectionSession);
+        String obProxyVersion = getObProxyVersion(connectionSession, isDirectedOBServer);
+        String obVersion = (String) connectionSession.getAttribute("OB_VERSION");
+        boolean isEnabledGlobalClientSession =
+                checkGlobalClientSessionEnabled(connectionSession, obProxyVersion, obVersion);
+        boolean isSupportedOracleModeKillSession = checkOracleModeKillSessionSupported(obVersion, connectionSession);
+        List<JdbcGeneralResult> finalResults = new ArrayList<>();
+        for (JdbcGeneralResult jdbcGeneralResult : results) {
             try {
                 jdbcGeneralResult.getQueryResult();
             } catch (Exception e) {
-                if (StringUtils.contains(e.getMessage(), "Unknown thread id")) {
-                    try {
-                        log.info("Kill query/session Unknown thread id error, try direct connect observer");
-                        directLinkServerAndExecute(sqlTuple.getExecutedSql(),
-                                connectionSession);
-                        return JdbcGeneralResult.successResult(sqlTuple);
-                    } catch (Exception e1) {
-                        log.warn("Failed to direct connect observer {}", e1.getMessage());
-                    }
+                if (isUnknownThreadIdError(e)) {
+                    jdbcGeneralResult = handleUnknownThreadIdError(connectionSession,
+                            jdbcGeneralResult, isDirectedOBServer,
+                            isEnabledGlobalClientSession, isSupportedOracleModeKillSession);
                 } else {
-                    log.warn("Failed to execute sql in kill session scenario, sqlTuple={}", sqlTuple, e);
+                    log.warn("Failed to execute sql in kill session scenario, sqlTuple={}",
+                            jdbcGeneralResult.getSqlTuple(), e);
                 }
             }
+            finalResults.add(jdbcGeneralResult);
+        }
+        return finalResults;
+    }
+
+    private static boolean checkObServerDirected(ConnectionSession connectionSession) {
+        String sql;
+        if (OB_MYSQL == connectionSession.getDialectType()) {
+
+            sql = "select PROXY_SESSID from oceanbase.gv$ob_processlist where ID =(select id from oceanbase.gv$ob_processlist where info like '%gv$ob_processlist%');";
+        } else {
+            sql = "select PROXY_SESSID from gv$ob_processlist where ID =(select id from gv$ob_processlist where info like '%gv$ob_processlist%');";
```

Comment:
both 2 query for mysql mode and oracle mode may got error result.

there may exits multiple connection link, e.g. one via directly connect, one via obproxy,
current condition may match more than one result, then queryForObject will throw exception.

Reply:
The result set will have only one record that determines whether the current session has an odp proxy on the link. so use queryForObject is ok

Reply:
But to get the original result set, and then make a unique judgment on the number of records, this is more robust, I will do that.